jdaviz/configs/default/plugins/model_fitting/tests/test_plugin.py

```diff
@@ -149,7 +149,8 @@ def test_register_cube_model(cubeviz_helper, spectrum1d_cube):
     modelfit_plugin._obj.cube_fit = True
     assert modelfit_plugin._obj.results_label_default == 'cube-fit model'
     assert modelfit_plugin._obj.results_label == test_label
-    with pytest.warns(AstropyUserWarning):
+    with warnings.catch_warnings():
+        warnings.simplefilter('ignore')
```

Comment:
If a ignore is still necessary if this is non-deterministic, it should be stricter, so at least please add a `category`.

Reply:
(although I'll also admit I'm a little confused why this warning is no longer raised, but is on main.... anyone have any ideas or have concerns this means that the behavior has changed?)

Reply:
But warning going away unexpectedly... worse case scenario is the fitting code is not being called at all... which might mean something else errored out and got sent to snackbar, which is not good. Did you try run this test case yourself in a notebook and see if it is behaving as expected?

Reply:
Warning still shows up in CI, so I guess that's good news?

```
        if model.linear:
>           warnings.warn(
                "Model is linear in parameters; consider using linear fitting methods.",
                AstropyUserWarning,
E               astropy.utils.exceptions.AstropyUserWarning: Model is linear in parameters; consider using linear fitting methods.

astropy\modeling\fitting.py:2145: AstropyUserWarning
```

Reply:
That is aggravating lol... the warnings somehow got captured in stderr ???

```
>           modelfit_plugin.calculate_fit()
E           Failed: DID NOT WARN. No warnings of type (<class 'astropy.utils.exceptions.AstropyUserWarning'>,) were emitted.
E           The list of emitted warnings is: [].

jdaviz/configs/default/plugins/model_fitting/tests/test_plugin.py:153: Failed
----------------------------- Captured stderr call -----------------------------
WARNING: Model is linear in parameters; consider using linear fitting methods. [astropy.modeling.fitting]
WARNING: Model is linear in parameters; consider using linear fitting methods. [astropy.modeling.fitting]
```

Reply:
and why with this PR, but not on others? 🤔

Reply:
In that case, for the sake of moving on, please use 

https://docs.python.org/3/library/warnings.html#warnings.filterwarnings

but explicitly define the category and message so we're sure we're only ignoring this warning that we expect.

Reply:
I still don't understand why the behavior is different between OSX and Windows jobs. Do you think this needs a follow-up tech debt investigation ticket? Or we just move on?

Your previous job: https://github.com/spacetelescope/jdaviz/actions/runs/5810800865/job/15752549545

Reply:
> and why with this PR, but not on others?

That is suspicious... Do you have a Mac? Just want to make sure this patch somehow doesn't break stuff on a Mac instead of Windows for once.

Reply:
I'm honestly not sure what to do here... I don't think the point of the test was to ensure the warning raised, but rather was just avoiding the warning... so a missing warning alone doesn't really concern me.  We do check that the resulting label exists in the data collection after, so I don't think its a sign of something going wrong, it just seems that they are being redirected on some systems (although I have no idea why we would see that in this PR and not others).